city_scrapers/spiders/il_lottery.py

```diff
@@ -0,0 +1,189 @@
+
+            if meeting_agenda_link is not None:
+                if meeting_minutes_link is not None:
+                    meeting['links'] = [meeting_agenda_link, meeting_minutes_link]
+                else:
+                    meeting['links'] = [meeting_agenda_link]
+            else:
+                meeting['links'] = []
+
+            yield meeting
+
+    def _parse_title(self, item):
+        """Parse or generate meeting title."""
+        suffix = ''
+        if 'QTR' in item:
+            suffix = "Quarterly "
+        elif 'Special' in item:
+            suffix = 'Special '
+        return "Lottery Control Board {}Meeting".format(suffix)
+
+    @staticmethod
+    def parse_time(source):
+        """Returns times given string with format h:mm am|pm"""
+        time_regex = re.compile(r'([1-9]):([0-5][0-9])\s?(am|pm)')
+        try:
+            hour, minute, period = time_regex.search(source).groups()
+
+            hour = int(hour)
+            minute = int(minute)
+            if (period == 'pm') and (hour != 12):
+                hour += 12
+            return time(hour, minute)
+        except AttributeError:
+            default_hour = 13
+            default_minute = 30
+            return time(default_hour, default_minute)
+
+    @staticmethod
+    def parse_day(source):
+        """Returns date"""
+        # search for dates with '/' (ex: 08/16/19)
+        if "/" in source:
+            date_match = re.search(r'(\d{2}\/\d{2}\/\d{2})', source).group()
+            dt = datetime.strptime(date_match, "%m/%d/%y").date()
+        # search for date in format "[month] [dd], [yyyy]" (ex: 'May 15, 2019')
+        else:
+            date_match = re.search(r"[a-zA-Z]{3,10} \d{1,2}?,? \d{4}", source).group()
+            dt = datetime.strptime(date_match, "%B %d, %Y").date()
+
+        return dt
+
+    def _parse_start(self, item):
+        """Parse start date and time."""
+        return datetime.combine(self.parse_day(item), self.parse_time(item))
+
+    def _parse_links(self, response, link_text_substring=''):
+        '''
+        Extracts link to agenda for a specific meeting date
+        Args:
+            date: A datetime object with the date of the desired agenda
+        Return:
+            List of dictionaries with the keys title (title/description of the link)
+            and href (link URL) for the agenda for the meeting on the requested date
+        '''
+        links = {}
+        for link in response.xpath('//p//a'):
+            link_text_selector = link.xpath('text()')
+            if link_text_selector:
+                link_text = link_text_selector.get()
+                if link_text_substring in link_text:
+                    link_date = self.parse_day(link_text)
+                    link_href = link.xpath('@href').get()
+                    links[link_date] = {
+                        "href": 'https://www.illinoislottery.com{}'.format(link_href),
```

Comment:
One shortcut you can use here is to create the full URL is `response.urljoin(link_href)` which automatically adds the domain of the response to any relative URLs

Reply:
Nifty, thanks!